### PR TITLE
Add FreeBSD case to gcenv.

### LINKED ISSUE #64935: GCMemoryInfo.MemoryLoadBytes is not implemented on FreeBSD
related to #14537 
Because of this, `GetGCMemoryInfo` test is failing. It is not clear if this impacts just the `GCMemoryInfo` and test or if that also impacts come other GC operations. 

This seems to come from:
 https://github.com/dotnet/runtime/blob/214a86e193176058b7fd1e4f50a537351764affb/src/coreclr/vm/gcenv.os.cpp#L913
and 
https://github.com/dotnet/runtime/blob/69b9000671f1b73b3fa17810a9f6cb31abb612a2/src/coreclr/gc/unix/gcenv.unix.cpp#L1143

and that eventually boils to 
https://github.com/dotnet/runtime/blob/69b9000671f1b73b3fa17810a9f6cb31abb612a2/src/coreclr/pal/src/misc/sysinfo.cpp#L432-L471

while `host_statistics()` does not seems to be supported on FreeBSD, we may be able to use `sysctlt`. I don't know exactly what and how but there seems to be info about general memory status. 

```
furt@d13:~/github/wfurt-runtime/src/coreclr $ sysctl -a | grep vm.stats.vm
vm.stats.vm.v_pdpages: 52746896
vm.stats.vm.v_tcached: 0
vm.stats.vm.v_cache_count: 0
vm.stats.vm.v_user_wire_count: 0
vm.stats.vm.v_free_severe: 7741
vm.stats.vm.v_interrupt_free_min: 2
vm.stats.vm.v_pageout_free_min: 34
vm.stats.vm.v_laundry_count: 58
vm.stats.vm.v_inactive_count: 231027
vm.stats.vm.v_inactive_target: 64605
vm.stats.vm.v_active_count: 1954
vm.stats.vm.v_wire_count: 959532
vm.stats.vm.v_free_count: 792657
vm.stats.vm.v_free_min: 12788
vm.stats.vm.v_free_target: 43070
vm.stats.vm.v_free_reserved: 2694
vm.stats.vm.v_page_count: 2018855
vm.stats.vm.v_page_size: 4096
vm.stats.vm.v_kthreadpages: 0
vm.stats.vm.v_rforkpages: 112322
vm.stats.vm.v_vforkpages: 608802
vm.stats.vm.v_forkpages: 792309
vm.stats.vm.v_kthreads: 23
vm.stats.vm.v_rforks: 1368
vm.stats.vm.v_vforks: 17614
vm.stats.vm.v_forks: 16655
vm.stats.vm.v_tfree: 79334088
vm.stats.vm.v_pfree: 34539982
vm.stats.vm.v_dfree: 2398725
vm.stats.vm.v_pdshortfalls: 58
vm.stats.vm.v_pdwakeups: 118
vm.stats.vm.v_reactivated: 684144
vm.stats.vm.v_intrans: 91070
vm.stats.vm.v_vnodepgsout: 271080
vm.stats.vm.v_vnodepgsin: 3647381
vm.stats.vm.v_vnodeout: 17159
vm.stats.vm.v_vnodein: 260117
vm.stats.vm.v_swappgsout: 767544
vm.stats.vm.v_swappgsin: 548901
vm.stats.vm.v_swapout: 44036
vm.stats.vm.v_swapin: 50815
vm.stats.vm.v_ozfod: 5063
vm.stats.vm.v_zfod: 63424950
vm.stats.vm.v_cow_optim: 255
vm.stats.vm.v_cow_faults: 6318711
vm.stats.vm.v_io_faults: 309807
vm.stats.vm.v_vm_faults: 72791650
```

Some of the entries _seems_ to match the `host_info_t`.

### PR DESCRIPTION
**Pull request by Thefrank**
Closes https://github.com/dotnet/runtime/issues/64935

### CODE REVIEW

**Comment by Thefrank** [comment]
Calculated based on https://github.com/dotnet/runtime/issues/64935#issuecomment-1031942707 and reworked from https://github.com/dotnet/runtime/issues/64935#issuecomment-1032071944. I used `vm.stats.vm.v_laundry_count` as `cache` always returned `0` for me

```
sysctl -d vm.stats.vm.v_inactive_count vm.stats.vm.v_free_count vm.stats.vm.v_laundry_count vm.stats.vm.v_cache_count
vm.stats.vm.v_inactive_count: Inactive pages
vm.stats.vm.v_free_count: Free pages
vm.stats.vm.v_laundry_count: Pages eligible for laundering
vm.stats.vm.v_cache_count: Dummy for compatibility
```
```
sysctl -h vm.stats.vm.v_inactive_count vm.stats.vm.v_free_count vm.stats.vm.v_laundry_count vm.stats.vm.v_cache_count
vm.stats.vm.v_inactive_count: 8,474,860
vm.stats.vm.v_free_count: 14,076,453
vm.stats.vm.v_laundry_count: 48,202
vm.stats.vm.v_cache_count: 0
```

Notes for future users:
- `sysconf` on FreeBSD is basically `sysctl` https://github.com/freebsd/freebsd-src/blob/e04c4b4a369df3f1dcbebbdf726193f02af60801/lib/libc/gen/sysconf.c however `sysconf` is generally more portable than `sysctl`
- I found that the code was mostly duplicated between src/coreclr/gc/unix/gcenv.unix.cpp and src/coreclr/pal/src/misc/sysinfo.cpp and added FreeBSD to both
- Reverts disabled test from https://github.com/dotnet/runtime/pull/64937

File: src/coreclr/pal/src/misc/sysinfo.cpp [hunk 1]
```diff
@@ -431,7 +431,39 @@ GlobalMemoryStatusEx(
     // We do this only when we have the total physical memory available.
     if (lpBuffer->ullTotalPhys > 0)
     {
-#ifndef __APPLE__
+#if defined (__APPLE__)
```
Comment (am11) [inline src/coreclr/pal/src/misc/sysinfo.cpp:434]: `GlobalMemoryStatusEx` in PAL is unused (outside PAL tests); good candidate of deletion:
i.e. everything this query reports: `git grep GlobalMemoryStatusEx ':/src/coreclr/pal'`

Comment (jkotas) [reply src/coreclr/pal/src/misc/sysinfo.cpp:434]: I have submitted the cleanup in separate PR: https://github.com/dotnet/runtime/pull/91554

@Thefrank Could you please revert changes in this file? It is dead code.

**Review by jkotas — APPROVED** [review]
LGTM. Thank you!

**Comment by Thefrank** [comment]
Updated title to reflect changed files.

@jkotas @am11 Thank you both for your continued assistance with community platforms!

**Pull request merged**